File: get_composio_integration.py
#!/usr/bin/env python3
import os
import sys
from dotenv import load_dotenv
from composio import ComposioToolSet

# Load environment variables
load_dotenv()

def verify_integration():
    """
    Verifies the Composio LinkedIn integration configuration
    
    This function fetches connected account details from Composio and verifies that:
    1. The connected account exists and is accessible
    2. The connected account is properly configured for LinkedIn
    3. The connected account has the necessary permissions
    
    Returns:
        bool: True if verification passes, False otherwise
    """
    try:
        # Get API key from environment variables or use default
        api_key = os.environ.get("COMPOSIO_API_KEY", "3kc5iwkpu3rtdk6cbvfuop")
        
        # Get connected account ID from environment variables or use default
        account_id = os.environ.get("COMPOSIO_ACCOUNT_ID", "624411e8-3384-4b2f-8fe8-4c835b81ea81")
        
        print(f"Verifying Composio LinkedIn connected account (ID: {account_id})...")
        
        # Initialize Composio client
        toolset = ComposioToolSet(api_key=api_key)
        
        # Get connected account details directly by ID
        connection = toolset.get_connected_account(account_id)
        
        if not connection:
            print(f"Connected account with ID {account_id} not found")
            return False
        
        # Log detailed connection information
        print("\n=== Composio LinkedIn Connected Account Details ===")
        print(f"Account ID: {connection.id if hasattr(connection, 'id') else 'Unknown'}")
        print(f"Name: {connection.name if hasattr(connection, 'name') else 'Unknown'}")
        print(f"Entity: {connection.entity if hasattr(connection, 'entity') else 'Unknown'}")
        print(f"Status: {connection.status if hasattr(connection, 'status') else 'Unknown'}")
        
        created_at = getattr(connection, 'createdAt', None)
        updated_at = getattr(connection, 'updatedAt', None)
        print(f"Created: {created_at if created_at else 'Unknown'}")
        print(f"Last Updated: {updated_at if updated_at else 'Unknown'}")
        
        # Verify LinkedIn-specific configuration
        if hasattr(connection, 'entity') and connection.entity.lower() != 'linkedin':
            print(f"ERROR: Connected account entity is \"{connection.entity}\" but expected \"linkedin\"")
            return False
        
        # Check if the connected account is active (case-insensitive)
        if hasattr(connection, 'status') and connection.status.lower() != 'active':
            print(f"ERROR: Connected account status is \"{connection.status}\" but expected \"active\"")
        else:
            print('✅ Connected account is active and properly configured for LinkedIn')
        
        # Check if the connected account has the expected organization URN
        expected_org_urn = os.environ.get("ORGANIZATION_URN")
        if expected_org_urn and hasattr(connection, 'organizationUrn') and expected_org_urn != connection.organizationUrn:
            print(f"WARNING: Organization URN mismatch. Expected: {expected_org_urn}, Actual: {connection.organizationUrn}")
        
        print("=== End of Connected Account Details ===\n")
        
        return True
    except Exception as e:
        print(f"Error verifying Composio LinkedIn connected account: {str(e)}")
        import traceback
        print(f"Stack trace: {traceback.format_exc()}")
        return False

if __name__ == "__main__":
    success = verify_integration()
    if success:
        print("Connected account verification completed successfully")
        sys.exit(0)
    else:
        print("Connected account verification failed")
        sys.exit(1)

File: get_composio_integration.js
import Composio from 'composio-core';
import dotenv from 'dotenv';

// Load environment variables
dotenv.config();

/**
 * Verifies the Composio LinkedIn integration configuration
 *
 * This function fetches integration details from Composio and verifies that:
 * 1. The integration exists and is accessible
 * 2. The integration is properly configured for LinkedIn
 * 3. The integration has the necessary permissions
 * 4. The integration matches the expected configuration in the .env file
 *
 * @returns {Promise<Object|null>} The integration details if successful, null otherwise
 */
async function getIntegrationDetails() {
  try {
    // Get API key from environment variables or use default
    const apiKey = process.env.COMPOSIO_API_KEY || "3kc5iwkpu3rtdk6cbvfuop";
    
    // Get integration ID from environment variables or use default
    const integrationId = process.env.COMPOSIO_INTEGRATION_ID || "53db8a77-df94-4544-b352-3c08a1110a37";
    
    console.log(`Verifying Composio LinkedIn integration (ID: ${integrationId})...`);
    
    // Initialize Composio client
    const composio = new Composio({ apiKey });
    
    // Get integration details
    const integration = await composio.integrations.get({
      integrationId
    });
    
    if (!integration) {
      console.error(`Integration with ID ${integrationId} not found`);
      return null;
    }
    
    // Log detailed integration information
    console.log("\n=== Composio LinkedIn Integration Details ===");
    console.log(`Integration ID: ${integration.id}`);
    console.log(`Name: ${integration.name}`);
    console.log(`Entity: ${integration.entity}`);
    console.log(`Status: ${integration.status || 'Unknown'}`);
    console.log(`Created: ${integration.createdAt ? new Date(integration.createdAt).toLocaleString() : 'Unknown'}`);
    console.log(`Last Updated: ${integration.updatedAt ? new Date(integration.updatedAt).toLocaleString() : 'Unknown'}`);
    
    // Verify LinkedIn-specific configuration
    if (integration.entity !== 'linkedin') {
      console.error(`ERROR: Integration entity is "${integration.entity}" but expected "linkedin"`);
      return null;
    }
    
    // Check if the integration has the necessary scopes/permissions
    const requiredScopes = ['w_member_social'];
    const missingScopes = [];
    
    if (integration.scopes) {
      console.log(`Scopes: ${integration.scopes.join(', ')}`);
      
      for (const scope of requiredScopes) {
        if (!integration.scopes.includes(scope)) {
          missingScopes.push(scope);
        }
      }
    } else {
      console.log('Scopes: Not available');
      missingScopes.push(...requiredScopes);
    }
    
    if (missingScopes.length > 0) {
      console.warn(`WARNING: Integration is missing required scopes: ${missingScopes.join(', ')}`);
    }
    
    // Verify the integration matches the expected configuration in .env
    const expectedOrgUrn = process.env.ORGANIZATION_URN;
    if (expectedOrgUrn && integration.organizationUrn && expectedOrgUrn !== integration.organizationUrn) {
      console.warn(`WARNING: Organization URN mismatch. Expected: ${expectedOrgUrn}, Actual: ${integration.organizationUrn}`);
    }
    
    // Check if the integration is active
    if (integration.status && integration.status !== 'active') {
      console.error(`ERROR: Integration status is "${integration.status}" but expected "active"`);
    } else {
      console.log('✅ Integration is active and properly configured for LinkedIn');
    }
    
    // Check if the integration has the expected tool name
    const expectedToolName = process.env.COMPOSIO_LINKEDIN_TOOL || 'LINKEDIN_MCP_CREATE_LINKED_IN_POST';
    if (integration.tools) {
      const hasExpectedTool = integration.tools.some(tool =>
        tool.name === expectedToolName || tool.id === expectedToolName
      );
      
      if (hasExpectedTool) {
        console.log(`✅ Integration has the expected tool: ${expectedToolName}`);
      } else {
        console.warn(`WARNING: Integration does not have the expected tool: ${expectedToolName}`);
        console.log('Available tools:', integration.tools.map(t => t.name || t.id).join(', '));
      }
    }
    
    console.log("=== End of Integration Details ===\n");
    
    return integration;
  } catch (error) {
    console.error("Error verifying Composio LinkedIn integration:", error);
    console.error("Stack trace:", error.stack);
    return null;
  }
}

// Execute the function
getIntegrationDetails()
  .then(integration => {
    if (integration) {
      console.log("Integration verification completed successfully");
    } else {
      console.error("Integration verification failed");
      process.exit(1);
    }
  })
  .catch(error => {
    console.error("Unexpected error during integration verification:", error);
    process.exit(1);
  });